main.py

```
@@ -1,98 +1,612 @@
+        )
+    except Exception as e:
+        print(f"Error generating report: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
+
+@app.post("/api/log-error")
+async def log_error(request: Request):
+    try:
+        error_data = await request.json()
+        print(f"[Error Log] {error_data.get('message', 'Unknown error')}")
+        return {"success": True}
+    except Exception:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)